rllib/core/rl_trainer/tests/test_trainer_runner.py

```diff
@@ -16,94 +14,84 @@
 )
 
 
-class TestTrainerRunner(unittest.TestCase):
-    @classmethod
-    def setUpClass(cls) -> None:
-        ray.init()
+REMOTE_SCALING_CONFIGS = {
+    "remote-cpu": TrainerScalingConfig(num_workers=1),
+    "remote-gpu": TrainerScalingConfig(num_workers=1, num_gpus_per_worker=0.5),
+    "multi-gpu-ddp": TrainerScalingConfig(num_workers=2, num_gpus_per_worker=1),
+    "multi-cpu-ddp": TrainerScalingConfig(num_workers=2, num_cpus_per_worker=2),
+    # "multi-gpu-ddp-pipeline": TrainerScalingConfig(
+    #     num_workers=2, num_gpus_per_worker=2
+    # ),
+}
 
-        # Settings to test
-        cls.scaling_configs = {
-            "local-cpu": TrainerScalingConfig(num_workers=0, num_gpus_per_worker=0),
-            "local-gpu": TrainerScalingConfig(num_workers=0, num_gpus_per_worker=0.5),
-            "remote-cpu": TrainerScalingConfig(num_workers=1),
-            "remote-gpu": TrainerScalingConfig(num_workers=1, num_gpus_per_worker=0.5),
-            "multi-gpu-ddp": TrainerScalingConfig(num_workers=2, num_gpus_per_worker=1),
-            "multi-cpu-ddp": TrainerScalingConfig(num_workers=2, num_cpus_per_worker=2),
-            # "multi-gpu-ddp-pipeline": TrainerScalingConfig(
-            #     num_workers=2, num_gpus_per_worker=2
-            # ),
-        }
-
-    @classmethod
-    def tearDownClass(cls) -> None:
-        ray.shutdown()
 
-    def test_trainer_runner_local(self):
+LOCAL_SCALING_CONFIGS = {
+    "local-cpu": TrainerScalingConfig(num_workers=0, num_gpus_per_worker=0),
+    "local-gpu": TrainerScalingConfig(num_workers=0, num_gpus_per_worker=0.5),
+}
 
-        tf1, tf, tfv = try_import_tf()
-        tf1.executing_eagerly()
 
-        # TODO (Avnish): tf does not clear out the GPU memory footprint, therefore
-        # doing it first before torch will result in OOM. Find a way to clear out the
-        # GPU memory footprint of tf.
-        fws = ["torch"]
-        scaling_modes = ["local-cpu", "local-gpu"]
-        test_iterator = itertools.product(fws, scaling_modes)
+class TestTrainerRunner(unittest.TestCase):
+    def setUp(self) -> None:
+        ray.init()
+
+    def tearDown(self) -> None:
+        ray.shutdown()
```

Comment:
this is good.

Reply:
oh you can totally shutdown ray with each test method. Good idea 👍